### PR TITLE
Give bodhi push a y flag

### LINKED ISSUE #2635: Give bodhi-push a -y flag so that it can be run in cron
We should give ```bodhi-push``` a -y flag that answers yes to the various questions so that it can be run in cron.

### PR DESCRIPTION
**Pull request by sebwoj**
fixed #2635

### CODE REVIEW

**Comment by Zlopez** [comment]
Shouldn't we just set `resume_all` and `resume` to `True` when `y` flag is added?

File: bodhi/server/push.py [hunk 1]
```diff
@@ -61,11 +61,12 @@ def update_sig_status(update):
 @click.option('--resume', help='Resume one or more previously failed pushes',
               is_flag=True, default=False)
 @click.option('--username', prompt=True)
+@click.option('--y', is_flag=True, default=False, help='Answers yes to the various questions')
```
Comment (bowlofeggs) [inline bodhi/server/push.py]:
```suggestion
@click.option('--yes/-y', is_flag=True, default=False, help='Answers yes to the various questions')
```

File: bodhi/server/push.py [hunk 2]
```diff
@@ -61,11 +61,12 @@ def update_sig_status(update):
...
 @click.version_option(message='%(version)s')
-def push(username, cert_prefix, **kwargs):
+def push(username, cert_prefix, y, **kwargs):
```
Comment (bowlofeggs) [inline bodhi/server/push.py]:
```suggestion
def push(username, cert_prefix, yes, **kwargs):
```

File: bodhi/server/push.py [hunk 3]
```diff
@@ -61,11 +61,12 @@ def update_sig_status(update):
...
     """Push builds out to the repositories."""
-    resume = kwargs.pop('resume')
-    resume_all = False
+    resume = kwargs.pop('resume') or y
```
Comment (bowlofeggs) [inline bodhi/server/push.py]:
```suggestion
    resume = kwargs.pop('resume') or yes
```

File: bodhi/server/push.py [hunk 4]
```diff
@@ -61,11 +61,12 @@ def update_sig_status(update):
...
+    resume_all = False or y
```
Comment (bowlofeggs) [inline bodhi/server/push.py]:
```suggestion
    resume_all = False or yes
```

File: bodhi/server/push.py [hunk 5]
```diff
@@ -148,8 +149,12 @@ def push(username, cert_prefix, **kwargs):
                 click.echo(update.title)
 
         if composes:
-            click.confirm('\n\nPush these {:d} updates?'.format(
-                sum([len(c.updates) for c in composes])), abort=True)
+            if y:
```
Comment (bowlofeggs) [inline bodhi/server/push.py]:
```suggestion
            if yes:
```

File: bodhi/tests/server/test_push.py [hunk 6]
```diff
@@ -471,6 +485,51 @@ def test_locked_updates(self, publish, mock_init):
             self.assertFalse(u.locked)
             self.assertIsNone(u.date_locked)
 
+    @mock.patch('bodhi.server.push.bodhi.server.notifications.init')
+    @mock.patch('bodhi.server.push.bodhi.server.notifications.publish')
+    def test_locked_updates_y_flag(self, publish, mock_init):
+        """
+        Test correct operation when there are some locked updates and --y flag is given.
```
Comment (bowlofeggs) [inline bodhi/tests/server/test_push.py]:
```suggestion
        Test correct operation when there are some locked updates and -y flag is given.
```

File: bodhi/tests/server/test_push.py [hunk 7]
```diff
@@ -471,6 +485,51 @@ def test_locked_updates(self, publish, mock_init):
...
+        ejabberd.locked = True
+        compose = models.Compose(
+            release=ejabberd.release, request=ejabberd.request, state=models.ComposeState.failed,
+            error_message=u'y r u so mean nfs')
+        self.db.add(compose)
+        self.db.commit()
+
+        with mock.patch('bodhi.server.push.transactional_session_maker',
+                        return_value=base.TransactionalSessionMaker(self.Session)):
+            result = cli.invoke(push.push, ['--username', 'bowlofeggs', '--y'])
```
Comment (bowlofeggs) [inline bodhi/tests/server/test_push.py]:
```suggestion
            result = cli.invoke(push.push, ['--username', 'bowlofeggs', '-y'])
```

**Review by bowlofeggs — CHANGES_REQUESTED** [review]
Cool, looks great! I'm going to change the flag to be ```--yes/-y``` instead of ```--y``` due to the convention of single-letter flags only using a single ```-```. I'll fix that, rebase your branch, and squash your commits and it should be good to merge. Thanks!

**Comment by bowlofeggs** [comment]
I'm applying this patch:

```
diff --git a/bodhi/server/push.py b/bodhi/server/push.py
index eca71623..455ed879 100644
--- a/bodhi/server/push.py
+++ b/bodhi/server/push.py
@@ -61,12 +61,12 @@ def update_sig_status(update):
 @click.option('--resume', help='Resume one or more previously failed pushes',
               is_flag=True, default=False)
 @click.option('--username', prompt=True)
-@click.option('--y', is_flag=True, default=False, help='Answers yes to the various questions')
+@click.option('--yes/-y', is_flag=True, default=False, help='Answers yes to the various questions')
 @click.version_option(message='%(version)s')
-def push(username, cert_prefix, y, **kwargs):
+def push(username, cert_prefix, yes, **kwargs):
     """Push builds out to the repositories."""
-    resume = kwargs.pop('resume') or y
-    resume_all = False or y
+    resume = kwargs.pop('resume') or yes
+    resume_all = False or yes

     initialize_db(config)
     db_factory = transactional_session_maker()
@@ -149,7 +149,7 @@ def push(username, cert_prefix, y, **kwargs):
                 click.echo(update.title)

         if composes:
-            if y:
+            if yes:
                 click.echo('\n\nPushing {:d} updates.'.format(
                     sum([len(c.updates) for c in composes])))
             else:
diff --git a/bodhi/tests/server/test_push.py b/bodhi/tests/server/test_push.py
index 4fed04f8..955c82d3 100644
--- a/bodhi/tests/server/test_push.py
+++ b/bodhi/tests/server/test_push.py
@@ -489,7 +489,7 @@ class TestPush(base.BaseTestCase):
     @mock.patch('bodhi.server.push.bodhi.server.notifications.publish')
     def test_locked_updates_y_flag(self, publish, mock_init):
         """
-        Test correct operation when there are some locked updates and --y flag is given.
+        Test correct operation when there are some locked updates and -y flag is given.
         """
         cli = CliRunner()
         # Let's mark ejabberd as locked and already in a push. bodhi-push should resume this
@@ -505,7 +505,7 @@ class TestPush(base.BaseTestCase):

         with mock.patch('bodhi.server.push.transactional_session_maker',
                         return_value=base.TransactionalSessionMaker(self.Session)):
-            result = cli.invoke(push.push, ['--username', 'bowlofeggs', '--y'])
+            result = cli.invoke(push.push, ['--username', 'bowlofeggs', '--yes'])

         self.assertEqual(result.exit_code, 0)
         mock_init.assert_called_once_with(active=True, cert_prefix=u'shell')
```

**Comment by bowlofeggs** [comment]
Ah, and apparently I didn't express the ```--yes``` and ```-y``` flags meaning the same thing correctly, here's another patch I'm applying:

```
diff --git a/bodhi/server/push.py b/bodhi/server/push.py
index 455ed879..f2262c0b 100644
--- a/bodhi/server/push.py
+++ b/bodhi/server/push.py
@@ -61,7 +61,8 @@ def update_sig_status(update):
 @click.option('--resume', help='Resume one or more previously failed pushes',
               is_flag=True, default=False)
 @click.option('--username', prompt=True)
-@click.option('--yes/-y', is_flag=True, default=False, help='Answers yes to the various questions')
+@click.option('--yes', '-y', is_flag=True, default=False,
+              help='Answers yes to the various questions')
 @click.version_option(message='%(version)s')
 def push(username, cert_prefix, yes, **kwargs):
     """Push builds out to the repositories."""
```

**Review by bowlofeggs — CHANGES_REQUESTED** [review]
This doesn't seem to work as I would expect. When I run ```bodhi-push``` without the ```-y``` flag, I am prompted to push a bunch of updates:

```
[vagrant@bodhi-dev bodhi]$ bodhi-push --username bowlofeggs
<snip>
Push these 230 updates? [y/N]:
```

When I run it with ```--yes```, I see this:

```
[vagrant@bodhi-dev bodhi]$ bodhi-push --username bowlofeggs --yes                               
There are no updates to push.
```

**Comment by Zlopez** [comment]
This looks like they aren't any composes found when `--yes` is provided, which is strange.

**Comment by bowlofeggs** [comment]
Excellent, this seems to work now. The f28-build failure is not because of this PR - I will rebase your PR to see if re-running CI gets that to work.

**Review by bowlofeggs — APPROVED** [review]

**Pull request merged**